01/components/TableFooter.js

```diff
@@ -0,0 +1,27 @@
+import React, { Component } from "react";
+import TableRow from "./TableRow";
+
+class TableFooter extends Component {
+  render() {
+    const { rows, summary } = this.props;
+    const tdJSX = rows.map((row) => {
+      const id = crypto.randomUUID();
```

Comment:
Trzeba uważać na kompatybilność:

![image](https://github.com/devmentor-pl/practice-react-composition/assets/55095257/88600474-279d-4bce-a679-6c6e3fec398e)


2022 to w cale nie tak dawno ;)

Dodatkowo wstawianie tego `key` w tym miejscu mija się z celem. Po każdym `render()` mamy inne ID więc nic ono nie daje (docelowo ma określać czy trzeba przeładować DOM). Zdecydowanie lepiej byłoby nadać to id na samej górze do danych wejściowych - wtedy react mógłby skorzystać z mechanizmy optymalizacji renderowania, bo to `id` nie będzie ciągle zmieniane :)

Reply:
czyli mam nie używać crypto?